### ISSUE TITLE
I am curious about the form of the infilling dataset for training.

### ISSUE THREAD

**Issue by ParksangwookA** [meta-llama/codellama#177]
Hello, I am studying machine learning and I have a question.

I'm not good at English, so I'm sorry for asking questions through a translator.

I am training codellama using the infilling dataset I created and SFTTrainer, and I am curious whether the infilling dataset is in the correct shape.

The current json data set format is

{"context" : "\<PRE\>{prefix}\<SUF\>{suffix}\<MID\>{answer}\<EOT\>"}

It is composed of a shape.

Also, what I'm curious about is whether the loss output when trained with the dataset is correct.

When using a general Trainer, I understand that loss is calculated by comparing the correct answer inferred by the model using the input labels.

Then, the dataset I am currently training with SFTrainer does not have labels.
There is only one "context", so I'm curious about what part of the loss is calculated from.

I would really appreciate it if you could leave any reply.

**Comment by ParksangwookA** [comment]
If the format of the data set mentioned above is incorrect,

{"input_ids" : "\<PRE\>{prefix}\<SUF\>{suffix}\<MID\>",
 "label" : "{answer}"
}

Is this the correct form?

thank you

**Comment by allenliu88** [comment]
@ParksangwookA did you have any ideas or solutions about this?
